src/Metadata/Resource/ResourceMetadataCollection.php

```diff
@@ -51,7 +51,9 @@ public function getOperation(?string $operationName = null, bool $forceCollectio
 
             foreach ($metadata->getOperations() ?? [] as $name => $operation) {
                 $isCollection = $operation instanceof CollectionOperationInterface;
-                if ('' === $operationName && \in_array($operation->getMethod() ?? HttpOperation::METHOD_GET, [HttpOperation::METHOD_GET, HttpOperation::METHOD_OPTIONS, HttpOperation::METHOD_HEAD], true) && ($forceCollection ? $isCollection : !$isCollection)) {
+                $method = $operation->getMethod() ?? HttpOperation::METHOD_GET;
+                $isGetOperation = HttpOperation::METHOD_GET === $method || HttpOperation::METHOD_OPTIONS === $method || HttpOperation::METHOD_HEAD === $method;
+                if ('' === $operationName && $isGetOperation && ($forceCollection ? $isCollection : !$isCollection)) {
```

Comment:
I'm taking this PR as an opportunity for a long time question I have: is adding an empty string as key in the `operationCache` a good idea?

Reply:
I need to prefix these, but yeah the representation of `null` as string would be an empty string ?

Reply:
I'm not sure to follow: if we ask a `null` `$operationName` to this `getOperation` method, it means we don't know the operation name. It it is cached, the same operation is always returned, even if the other conditions have changed (`$isGetOperation`, `$forceCollection`...). It seems dangerous to me.

Reply:
Let's fix this in another patch then.